torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -933,21 +934,64 @@ def _run_param_init_fn():
         self._hook_registered = False
 
     def _init_param_exec_order_wrap_policy(self, *args, **kwargs) -> None:
+        auto_wrap_policy = kwargs["auto_wrap_policy"]
+        module = kwargs["module"]
+        if isinstance(
+            auto_wrap_policy.tracing_config,
+            TracingConfig
+        ):
+            tracer = auto_wrap_policy.tracing_config.tracer
+            execution_info = _init_execution_info(module)
+            with _patch_tracer(
+                tracer=tracer,
+                root_module=module,
+                execution_info=execution_info,
+            ):
+                try:
+                    tracer.trace(module, auto_wrap_policy.tracing_config.concrete_args)
```

Comment:
can we trace through meta modules / modules initialized with fake tensors? That might be pretty useful if all we'd like to do is understand the execution order, and defer actual initialization.

Reply:
@rohan-varma Can you provide some links to get context on meta modules and fake tensors?

Reply:
The FakeTensor and deferred module init sections here: https://github.com/pytorch/torchdistx#getting-started should provide some useful context.